ios/browser/api/webcompat_reporter/webcompat_reporter_service_factory_wrapper.h

```diff
@@ -0,0 +1,22 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_IOS_BROWSER_API_WEBCOMPAT_REPORTER_WEBCOMPAT_REPORTER_SERVICE_FACTORY_WRAPPER_H_
+#define BRAVE_IOS_BROWSER_API_WEBCOMPAT_REPORTER_WEBCOMPAT_REPORTER_SERVICE_FACTORY_WRAPPER_H_
+
+#import <Foundation/Foundation.h>
+
+#include "keyed_service_factory_wrapper.h"  // NOLINT
+
+@protocol WebcompatReporterWebcompatReporterHandler;
+
+OBJC_EXPORT
+NS_SWIFT_NAME(WebcompatReporter.ServiceFactory)
+@interface WebcompatReporterServiceFactory
+    : KeyedServiceFactoryWrapper <
+          id <WebcompatReporterWebcompatReporterHandler>>
```

Comment:
please don't change type names, to differentiate between the service and the ios wrapper use the IOS suffix so this should be `WebcompatReporterServiceIOS`

Reply:
also the service itself should be `WebcompatReporterServiceImpl` so the mojo interface can be `WebcompatReporterService`